### ISSUE TITLE
Remove windows build support

### ISSUE THREAD

**Issue by adfoster-r7** [rapid7/mettle#247]
Placeholder ticket to decide if we want to remove windows build support from Mettle.

At the minute the Windows support isn't fully implemented, and we're currently paying the maintenance cost for windows support when CI builds break - but it's not functionality that the average user is aware of. 

Do we want to remove this functionality and declare that Mettle is not intended to support windows?

**Comment by adfoster-r7** [comment]
@sempervictus Are you currently using mettle on Windows at all?

**Comment by zeroSteiner** [comment]
Four out of my last 5 PRs either broke the build or had to dance around it with the precompiled.

* https://github.com/rapid7/mettle/pull/230/files#diff-10d3a8489fe9a5b40b9ddbf28d041fd6e0dc4cd85ec9585dbc6c68394321d39f
* https://github.com/rapid7/mettle/pull/228/files
* https://github.com/rapid7/mettle/pull/225#issuecomment-925344377
* https://github.com/rapid7/mettle/pull/216#issuecomment-856959862

It was also an issue on this PR from a community member I reviewed: https://github.com/rapid7/mettle/pull/243#issuecomment-1448194089

We also have two open issues regarding the MinGW builds:
* #208 
* #101

**Comment by adfoster-r7** [comment]
> i have a "re-consolidate meterp on Rust" effort on my TODO list

I have a Rust Meterpreter PoC working from a year ago; but I'm not sure if we'd want the maintenance overhead of another Meterpreter that doesn't quite have the same semantics as the other implementations

**Comment by sempervictus** [comment]
I think the push to Rust would actually ease maintenance - all the cool kids are learning _that_ these days, not C.
Clean-sheet implementation at current state of industry should give you something like a Brute Rattel (switching semantcs from syscalls+ROP to WinAPI calls with and without ROP on the fly, mem obfu, unhooking, built & runtime obfuscation, etc), allow us to re-think how session state is managed to handle C2+ better, and eventually deprecate the C versions (though, IIRC Adam's point regarding the payload _size_ of a Rust meterp compared to Mettle's still stands - Rust's runtime does bring in some overhead so a `nostd` approach might be the way to go, and that's a "ground-up" decision to be made).

Any chance you might be able to publish whatever you've got? Would make a hell of a springboard for a GSoC effort or tinker-toy for payload writers to expand upon in copious "spare" time :smile:.

**Comment by Ishaanahuja7** [comment]
@adfoster-r7 I agree with @sempervictus If you could publish the code in a repo somewhere, the community can help achieve a working rust meterpreter faster since number of rust developers are growing exponentially faster than C/C++ developers now + its become a somewhat faster and cleaner alternative to C/C++ nowadays (Look a ripgrep and fd).
